src/chromeDebugAdapter.ts

```diff
@@ -403,3 +431,42 @@ function findExecutable(program: string): string | undefined {
 
     return undefined;
 }
+
+async function findNewlyLaunchedChromeProcess(semaphoreFile: string): Promise<string> {
+    const regexPattern = /processid\s+=\s+(\d+)\s*;/i
+    let lastAccessFileContent: string;
+    for (let i = 0 ; i < 25; i++) {
+        if (fs.existsSync(semaphoreFile)) {
+            lastAccessFileContent = fs.readFileSync(semaphoreFile, {
+                encoding: 'utf16le'
+            }).toString();
+
+            const lines = lastAccessFileContent.split("\n");
+
+            const matchedLines = (lines || []).filter(line => line.match(regexPattern))
+            if (matchedLines.length > 1) {
+                throw new Error(`Unexpected semaphore file format ${lines}`);
+            }
+
+            if (matchedLines.length === 1) {
+                const match = matchedLines[0].match(regexPattern);
+                return match[1];
+            }
+            // else == 0, wait for 200 ms delay and try again.
+        }
+        await new Promise<void>((resolve) => {
+            setTimeout(resolve, 200);
+        })
+    }
+
+    const error = new Error(`Cannot acquire Chrome process id`);
+    let telemetryProperties: any = {
+        semaphoreFileContent: lastAccessFileContent
+    };
+
+    coreUtils.fillErrorDetails(telemetryProperties, error);
+
+    telemetry.telemetry.reportEvent('error', telemetryProperties);
+
+    return null;
```

Comment:
Is this supposed to just keep going and try to attach if it can't find the Chrome pid? Or should it throw and fail the launch?

Reply:
Yes, after evaluated two approaches, we reached to a conclusion that we should let the debugging session go as much as it can. Not being able to find process id doesn't necessary mean Chrome is not launched nor non-debuggable. One known side-effect is merely that Chrome can't be closed at the end of the session. Due to current multi-debugger-client design of Chrome, that is not an issue any more.